packages/server/graphql/public/mutations/batchArchiveTasks.ts

```diff
@@ -0,0 +1,33 @@
+import {getUserId} from '../../../utils/authorization'
+
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import publish from '../../../utils/publish'
+
+import {MutationResolvers} from '../resolverTypes'
+import isValid from '../../isValid'
+import archiveTasksForDB from '../../../safeMutations/archiveTasksForDB'
+
+const batchArchiveTasks: MutationResolvers['batchArchiveTasks'] = async (
+  _source,
+  {taskIds},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const viewerId = getUserId(authToken)
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  // VALIDATION
+  const tasks = (await dataLoader.get('tasks').loadMany(taskIds)).filter(isValid)
+  const validTasks = tasks.filter(({createdBy}) => createdBy === viewerId)
+  const validTaskIds = validTasks.map(({id}) => id)
+
+  // RESOLUTION
+  archiveTasksForDB(validTasks)
+
+  validTaskIds.forEach((taskId) => {
+    publish(SubscriptionChannel.TASK, taskId, 'BatchArchiveTasksSuccess', {taskId}, subOptions)
```

Comment:
-1 the source `{taskId}` does not match the `BatchArchiveTasksSuccessSource` which defines `taskIds: string[]`.
As the error does not show up in usage, I assume we might not subscribe to this event correctly, which would mean another team member wouldn't receive the archive update.

Reply:
I changed it to be a team level subscription so when you archive all tasks on your board, or your team's board, others will see changes too